### DISCUSSION TITLE
How GMM Algorithm Optimization Is Applied in the Framework

### DISCUSSION THREAD

**Discussion by SaturnWorld** [deepmodeling/deepmd-kit#792] [Q&A]
Dear  developers and users:
        In the paper 《Pushing the limit of molecular dynamics with ab initio accuracy to 100 million atoms with machine learning》,i find several optimizations for embedding net computations , such as
        1. replace MATMUL and ADD operations with GEMM
        2. replace CONCAT and ADD operations with GEMM
        However, I do not find the relevant code implementation in the source code of the deepmd-kit framework. How does the framework apply this optimization solution? Thanks.

**Comment by tuoping** [comment]
@denghuilu

**Reply by denghuilu** [reply]
These optimizations did not apply to the deepmd-kit source code yet. The main reason is that we now have a better alternative, please see our introduction to model compression in the documentation. For the optimization of neural network structure, model compression can bring better speedups. You can enable this new feature in the deepmd-kit-v2.0 releases.

**Reply by denghuilu** [reply]
If you are interested in the paper's implementation, you can access the code by https://drive.google.com/file/d/1Rk42sTLHETIMD-qAdqTE0mQdM4vaVMJj/view

**Reply by SaturnWorld** [reply]
Thank you very much~
